### PR TITLE
[IMP] mail: make chatter attachment button more explicit

### PR DESCRIPTION
**Pull request by seb-odoo**
task-2413814

### CODE REVIEW

File: addons/mail/static/src/components/chatter_topbar/chatter_topbar.js [hunk 1]
```diff
@@ -40,6 +40,23 @@ class ChatterTopbar extends Component {
...
+            !this.chatter ||
+            !this.chatter.thread ||
+            this.chatter.thread.allAttachments.length === 0
+        ) {
+            return this.env._t("0 Attachment");
+        }
+        if (this.chatter.thread.allAttachments.length === 1) {
+            return this.env._t("1 Attachment");
+        }
+        return _.str.sprintf(this.env._t("%s Attachments"), this.chatter.thread.allAttachments.length);
```
Comment (seb-odoo) [inline addons/mail/static/src/components/chatter_topbar/chatter_topbar.js:57]: still todo make lowercase, todo after rebase on https://github.com/odoo/odoo/pull/63320/

**Review by alexkuhn — CHANGES_REQUESTED** [review]
There's too many component and flow-specific logic in this PR, which makes things hard to understand and overall code quite fragile.

Instead of the `isDoAddAttachment` approach, I would go with the `isFileBrowserOpen` one. I think it will require some event on focus on file input to detect whether file browser is open or not. This approach is much more declarative and better overall.

**Comment by seb-odoo** [comment]
> There's too many component and flow-specific logic in this PR, which makes things hard to understand and overall code quite fragile.
> 
> Instead of the `isDoAddAttachment` approach, I would go with the `isFileBrowserOpen` one. I think it will require some event on focus on file input to detect whether file browser is open or not. This approach is much more declarative and better overall.

This would require to make separate model for each composer, and also require to make models for attachment box, which I tried to avoid in the context of this PR. But I guess it might be time, the 1-1 thread-composer is really annoying in many tasks.

**Comment by alexkuhn** [comment]
> This would require to make separate model for each composer, and also require to make models for attachment box, which I tried to avoid in the context of this PR. But I guess it might be time, the 1-1 thread-composer is really annoying in many tasks.

I'm afraid the current approach of PR would make necessary changes to our code very hard. Considering it's not a FIX, this is why it's better not to rush with solution that would code lots of issues.

About 1 record per composer in view, we must keep in mind that components is the painting of models. Just wanted to insist the code shouldn't work as _"each mounted component maps to a record"_, but it's _"context of a thread being viewed, there's some composer concern"_. So basically the models `ThreadView`/`ThreadViewer` are almost what we need... Maybe clean a bit to nicely pack composer concerns in a view, but that's a causal one2one relationship in a nutshell.

**Comment by seb-odoo** [comment]
Better redo the task from scatch based on our recent code.

**Pull request closed without merging**